From: Jeff Mahoney <jeffm@suse.com>
Subject: [PATCH] ipmi: Fix section type conflict with hotmod_handler

 Module parameters can't be static since the module macros explicitly put
 those symbols in the __param section.

Signed-off-by: Jeff Mahoney <jeffm@suse.com>
---

 drivers/char/ipmi/ipmi_si_intf.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)
--- a/drivers/char/ipmi/ipmi_si_intf.c
+++ b/drivers/char/ipmi/ipmi_si_intf.c
@@ -1098,7 +1098,7 @@ static unsigned int num_slave_addrs;
 #define IPMI_MEM_ADDR_SPACE 1
 static char *addr_space_to_str[] = { "i/o", "mem" };
 
-static int hotmod_handler(const char *val, struct kernel_param *kp);
+int hotmod_handler(const char *val, struct kernel_param *kp);
 
 module_param_call(hotmod, hotmod_handler, NULL, NULL, 0200);
 MODULE_PARM_DESC(hotmod, "Add and remove interfaces.  See"
@@ -1514,7 +1514,7 @@ static int check_hotmod_int_op(const cha
 	return 0;
 }
 
-static int hotmod_handler(const char *val, struct kernel_param *kp)
+int hotmod_handler(const char *val, struct kernel_param *kp)
 {
 	char *str = kstrdup(val, GFP_KERNEL);
 	int  rv;
